code/game/objects/items/weaponry.dm

```diff
@@ -505,8 +505,90 @@ for further reading, please see: https://github.com/tgstation/tgstation/pull/301
 	attack_verb_continuous = list("bludgeons", "whacks", "disciplines", "thrashes")
 	attack_verb_simple = list("bludgeon", "whack", "discipline", "thrash")
 
+/obj/item/cane/examine(mob/user, thats)
+	. = ..()
+	. += span_notice("This item can be used to support your weight, preventing limping from any broken bones on your legs you may have.")
+
+/obj/item/cane/equipped(mob/living/user, slot, initial)
+	. = ..()
+	movement_support_add(user, slot, initial)
+
+/obj/item/cane/dropped(mob/living/user, slot, initial)
+	. = ..()
+	movement_support_del(user, slot, initial)
+
+/obj/item/cane/proc/movement_support_add(mob/living/user, slot, initial)
+	if(!(slot & ITEM_SLOT_HANDS))
+		return
+	RegisterSignal(user, COMSIG_MOB_LIMP_CHECK, PROC_REF(handle_limping))
+	user.set_usable_legs()
+	return TRUE
+
+/obj/item/cane/proc/movement_support_del(mob/living/user, slot, initial)
+	if((slot & ITEM_SLOT_HANDS))
+		return
+	UnregisterSignal(user, list(COMSIG_MOB_LIMP_CHECK))
+	user.set_usable_legs()
+	return TRUE
+
+/obj/item/cane/proc/handle_limping(mob/living/user)
+	return COMPONENT_CANCEL_LIMP
+
+/obj/item/cane/crutch
+	name = "medical crutch"
+	desc = "A medical crutch used by people missing a leg. Not all that useful if you're missing both of them, though."
+	icon = 'icons/obj/weapons/staff.dmi'
+	icon_state = "crutch_med"
+	inhand_icon_state = "crutch_med"
+	lefthand_file = 'icons/mob/inhands/weapons/melee_lefthand.dmi'
+	righthand_file = 'icons/mob/inhands/weapons/melee_righthand.dmi'
+	force = 12
+	throwforce = 8
+	w_class = WEIGHT_CLASS_BULKY
+	custom_materials = list(/datum/material/iron = SMALL_MATERIAL_AMOUNT * 0.5)
+	attack_verb_continuous = list("bludgeons", "whacks", "thrashes")
+	attack_verb_simple = list("bludgeon", "whack", "thrash")
+
+/obj/item/cane/crutch/examine(mob/user, thats)
+	. = ..()
+	// tacked on after the cane string
+	. += span_notice("As a crutch, it can also help lessen the slowdown incurred by missing a leg.")
+
+/obj/item/cane/crutch/movement_support_add(mob/living/user, slot, initial)
+	. = ..()
+	if(!.)
+		return
+	RegisterSignal(user, COMSIG_LIVING_LIMBLESS_SLOWDOWN, PROC_REF(handle_slowdown))
+	user.set_usable_legs()
+	user.AddElementTrait(TRAIT_WADDLING, REF(src), /datum/element/waddling)
+
+/obj/item/cane/crutch/movement_support_del(mob/living/user, slot, initial)
+	. = ..()
+	if(!.)
+		return
+	UnregisterSignal(user, list(COMSIG_LIVING_LIMBLESS_SLOWDOWN, COMSIG_MOB_LIMP_CHECK))
+	REMOVE_TRAIT(user, TRAIT_WADDLING, REF(src))
+
+/obj/item/cane/crutch/proc/handle_slowdown(mob/living/user, limbless_slowdown)
+	SIGNAL_HANDLER
+	var/leg_amount = user.usable_legs
+	// Don't do anything if the number is equal (or higher) to the usual.
+	if(leg_amount >= user.default_num_legs)
+		return
+	// If we have at least one leg and it's less than the default, reduce slowdown by 60%.
+	if(leg_amount && (leg_amount < user.default_num_legs))
+		limbless_slowdown *= 0.4
+	return limbless_slowdown
```

Comment:
Returning numbers from signals is a big no-no. Huge no-no. 

When I did this for another server I just had it return a list of modifiers to the final slowdown number. 

![image](https://github.com/tgstation/tgstation/assets/51863163/c832e136-f68e-4c8b-95fb-cd98a6ea24c2)